tests/test_scheduler.py

```
@@ -0,0 +1,527 @@
+        
+        # Get predictions from each tree individually
+        pred_tree0 = self._get_tree_predictions(model, 'cpu', 0, 1)
+        pred_tree1 = self._get_tree_predictions(model, 'cpu', 1, 2)
+        
+        # Tree 0 should have lr=0.55 applied
+        # For samples with X > 0 (indices 2,3): raw gradient points toward 1.0
+        # The tree fits gradient -0.5 (half of 1.0) for these, scaled by 0.55 => 0.55 * -(-0.5 * 2) = 0.55
+        # For simplicity, just verify the predictions are what we expect:
+        # Sample 2,3 should have prediction ~0.55 (lr * leaf_value)
+        
+        print(f"\nCPU Linear Scheduler Test:")
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)